queries/tags.scm

```diff
@@ -0,0 +1,33 @@
+(classlessPredicate
+  name: (predicateName) @name
+  (#not-eq? @name "")) @definition.function
```

Comment:
@maxbrunsfeld I'm not sure about all these predicates, but semantic has a `Parse.success` matcher for [these](https://github.com/github/semantic/blob/master/semantic-codeql/src/Language/CodeQL/Tags.hs#L121). Is this the right way to do it?

Reply:
I don't think we need these predicates. In all of these cases, I don't think the grammar will allow an empty string.

Reply:
If I don't do this, something like: `NoCall()` will tag as:

```
NoCall    	 | type    	ref (15, 0) - (15, 6) `NoCall()`
          	 | function	def (15, 0) - (15, 8) `NoCall()`
```

Reply:
Oh, interesting.

It looks like that's because there's a parse error there, but the CLI isn't showing it (because it's a `missing` node that is hidden). Tree-sitter should probably surface that error better. But for the case of tags, I've fixed it via https://github.com/tree-sitter/tree-sitter/commit/ba70927f573b0d098046da77888d3219ee31cc9d.